paasta_tools/cli/cmds/status.py

```diff
@@ -1331,24 +1342,35 @@ def get_replica_state(pod: KubernetesPodV2) -> ReplicaState:
             # TODO: Take sidecar containers into account
             #   This logic likely needs refining
             state = ReplicaState.UNREACHABLE
```

Comment:
Whoops, you are right. I broke this logic (and I see I left a bunch of extraneous print statements). The intent was originally to only detect conditions WARMING_UP/STARTING/crashloopbackoff here and anything else that is not `pod.ready` meant it was UNREACHABLE. 

Where I am currently just returning WARMING_UP if main_container has no recent restarts, only if the container is < healthcheck_grace_period seconds old will we be in WARMING_UP, otherwise UNREACHABLE.

Reply:
@qui would you mind looking at this logic again? It still feels a little clunky, but now ensures we identify things in UNREACHABLE if we have no restarts and are not warming up.

Should we be able to get into a state where restarts are > 0 but the state really should be UNREACHABLE?